chore: Replace manually generated clients with autogenerated

josipbagaric opened the pull request:
## Changes

### Backend
- Auto-generate client libraries for
    - `@energyweb/exchange` = **New** package `@energyweb/exchange-client`
    - `@energyweb/origin-backend` = **Repurposed** the package `@energyweb/origin-backend-client`
    - `@energyweb/origin-organization-irec-api` = **New** package `@energyweb/origin-organization-irec-api-client`
- **Rich Swagger definitions**
- For now, move `Configuration.Entity` to the device registry, as it's only used there now (and solves some circular dependencies
- Add a prefix `exchange_` for Exchange DB table names
- **Deleted** the `@energyweb/origin-backend-client-mocks` package

### UI
- Rename `OffChainDataSource` to `BackendClient`
- Make use of the new auto-generated clients


### TO-DO
- [x] Fix file upload progress bar
- [ ]  ~Fix file upload cancel token~ (no obvious solution to this yet 🤷 )

ioncreature commented on packages/device-registry/src/test/DeviceFacade.test.ts:
```diff
@@ -60,7 +59,7 @@ describe('Device Facade', () => {
         });
 
-        it('should log a new meter reading', async () => {
+        xit('should log a new meter reading', async () => {
```
Why do you skip these tests?

josipbagaric replied:
Together with @kosecki123, we've agreed on a workflow to open a **work-in-progress** PR as soon as possible when we are working on features, so that we can track progress.

To make the basic build run, I tend to comment failing tests in work-in-progress PRs, and then I fix them before I request for a PR review from the team.

That's just my workflow.

josipbagaric committed:
chore(utils-general): remove Configuration

BREAKING CHANGE: Configuration is no longer a part of utils-general

josipbagaric committed:
chore(device-registry): Move Configuration to device-registry

josipbagaric committed:
chore(origin-backend-client-mocks): remove the package

BREAKING CHANGE: The package is no longer used, and will be removed from the repo

josipbagaric committed:
chore(origin-backend-core): remove unused backend client definitions

josipbagaric committed:
chore(origin-backend): add rich Swagger definitions

josipbagaric committed:
feat(exchange): rich Swagger definitions + add prefix to all exchange tables

josipbagaric committed:
feat(exchange-client): create an autogenerated Exchange client

josipbagaric committed:
feat(origin-backend-client): Make origin backend client autogenerated

BREAKING CHANGE: The Origin backend client now uses a totally different API

josipbagaric committed:
chore(issuer-api): use HttpStatus enum instead of number statuses

josipbagaric committed:
chore(origin-organization-irec-api): add rich Swagger definitions

josipbagaric committed:
feat(origin-organization-irec-api-client): create an autogenerated client for origin-organization-irec-api

josipbagaric committed:
chore(origin-backend-app): show all package versions

josipbagaric committed:
chore: replace manually created API clients with autogenerated

josipbagaric committed:
chore: update solar sim

josipbagaric committed:
chore: change build scripts for autogenerated API clients

josipbagaric committed:
chore(origin-backend-client): remove index.ts from the repo

josipbagaric committed:
Merge branch 'master' into chore/remove-manually-created-clients

josipbagaric committed:
chore: Add Authentication to the Swagger UI

kosecki123 commented on packages/origin-backend/migrations/1604562378046-SwaggerChanges.ts:
```diff
@@ -0,0 +1,19 @@
+        await queryRunner.query(`ALTER TABLE "organization_invitation" DROP COLUMN "status"`);
+        await queryRunner.query(
+            `ALTER TABLE "organization_invitation" ADD "status" character varying NOT NULL`
+        );
```
This will fail on existing environments if a drop won't happen.

josipbagaric replied:
Yeah, we need a default.

josipbagaric replied:
I actually didn't change this. Someone else forgot to update migrations. 🤷

kosecki123 commented on packages/exchange/migrations/1604562715444-ExchangeTablePrefix.ts:
```diff
@@ -0,0 +1,91 @@
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(
+            `CREATE TABLE "exchange_account" ("createdAt" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "updatedAt" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), "id" uuid NOT NULL DEFAULT uuid_generate_v4(), "userId" character varying NOT NULL, "address" character varying NOT NULL, CONSTRAINT "PK_6dbfd7bdb34f40b831d3013a57f" PRIMARY KEY ("id"))`
+        );
```
We are creating a new set of tables here, could we try `ALTER TABLE table_name
RENAME TO new_table_name;` instead?

josipbagaric replied:
Good point. This is the autogenerated code by NestJS, not sure why they don't already do it that way.

josipbagaric committed:
chore: update migrations

josipbagaric committed:
chore(exchange): fix tests
